client/extensions/woocommerce/state/ui/payments/methods/test/selectors.js

```diff
@@ -0,0 +1,415 @@
+		} );
+
+		it( 'should return the WC-API methods list if there are no edits in the state', () => {
+			const state = {
+				extensions: {
+					woocommerce: {
+						sites: {
+							123: {
+								paymentMethods: [
+									{ id: 1, name: 'Method1' },
+									{ id: 2, name: 'Method2' },
+								],
+							},
+						},
+						ui: {},
+					},
+				},
+				ui: {
+					selectedSiteId: 123,
+				},
+			};
+
+			expect( getPaymentMethodsWithEdits( state ) ).to.deep.equal( [
+				{ id: 1, name: 'Method1' },
+				{ id: 2, name: 'Method2' },
+			] );
+		} );
+
+		it( 'should apply the "edits" changes to the method list', () => {
+			const state = {
+				extensions: {
+					woocommerce: {
+						sites: {
+							123: {
+								paymentMethods: [
+									{ id: 1, name: 'Method1' },
+									{ id: 2, name: 'Method2' },
+									{ id: 3, name: 'Method3' },
+								],
+							},
+						},
+						ui: {
+							payments: {
+								123: {
+									methods: {
+										creates: [
+											{ id: { index: 0 }, name: 'Method4' },
+										],
+										updates: [
+											{ id: 2, name: { value: 'EditedMethod2' } },
+										],
+										deletes: [
+											{ id: 1 },
+										],
+										currentlyEditingId: null,
+									},
+								},
+							},
+						},
+					},
+				},
+				ui: {
+					selectedSiteId: 123,
+				},
+			};
+
+			expect( getPaymentMethodsWithEdits( state ) ).to.deep.equal( [
+				{ id: 2, name: 'EditedMethod2' },
+				{ id: 3, name: 'Method3' },
+				{ id: { index: 0 }, name: 'Method4' },
+			] );
+		} );
+
+		it( 'should NOT apply the uncommited changes made in the modal', () => {
+			const state = {
+				extensions: {
+					woocommerce: {
+						sites: {
+							123: {
+								paymentMethods: [
+									{ id: 1, name: 'Method1' },
+								],
+							},
+						},
+						ui: {
+							payments: {
+								123: {
+									methods: {
+										creates: [],
+										updates: [],
+										deletes: [],
+										currentlyEditingId: 1,
+										currentlyEditingChanges: { name: 'This name has not been saved yet' },
+									},
+								},
+							},
+						},
+					},
+				},
+				ui: {
+					selectedSiteId: 123,
+				},
+			};
+
+			expect( getPaymentMethodsWithEdits( state ) ).to.deep.equal( [ { id: 1, name: 'Method1' } ] );
+		} );
+	} );
+
+	describe( 'getCurrentlyEditingPaymentMethod', () => {
+		it( 'should return null when there is no method being edited', () => {
+			const state = {
+				extensions: {
+					woocommerce: {
+						sites: {
+							123: {
+								paymentMethods: [
+									{ id: 1 },
+								],
+							},
+						},
+						ui: {
+							payments: {
+								123: {
+									methods: {
+										currentlyEditingId: null,
+									},
+								},
+							},
+						},
+					},
+				},
+				ui: {
+					selectedSiteId: 123,
+				},
+			};
+
+			expect( getCurrentlyEditingPaymentMethod( state ) ).to.be.null;
+			expect( isCurrentlyEditingPaymentMethod( state ) ).to.be.false;
```

Comment:
It looks like you're testing two selectors in one test here — each `describe` block should only test one selector